src/main/java/pl/edu/agh/firecell/core/Window.java

```diff
@@ -17,26 +18,32 @@
 import pl.edu.agh.firecell.core.io.IOListener;
 import pl.edu.agh.firecell.core.io.KeyEvent;
 import pl.edu.agh.firecell.core.statebuilder.Room;
-import pl.edu.agh.firecell.core.statebuilder.RoomStorage;
+import pl.edu.agh.firecell.storage.RoomStorage;
 import pl.edu.agh.firecell.core.statebuilder.StateBuilderScene;
 import pl.edu.agh.firecell.core.util.LoggingOutputStream;
+import pl.edu.agh.firecell.core.util.StateUtils;
 import pl.edu.agh.firecell.model.SimulationConfig;
+import pl.edu.agh.firecell.storage.SimulationStorage;
 
 import java.io.IOException;
 import java.io.PrintStream;
 import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
 import java.util.Objects;
 
 import static org.lwjgl.glfw.GLFW.*;
 import static org.lwjgl.opengl.GL11.*;
 
-public class Window {
+public class Window implements StartSimulationHandler {
+    private static final Path SAVED_SIMULATIONS_PATH = Path.of("simulations");
+    private static final String STATES_DIRECTORY_NAME = "states";
```

Comment:
Do we really need to modify the most outer class?
It seems to be really poorly placed, some scene should handle such special mechanics, and if they have to be shared between scenes, they can be passed in some handlers, as previously it was done using `start-` and `finishSimulationHandler`